map/chart_generator.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "indexer/feature_altitude.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "std/deque.hpp"
+#include "std/vector.hpp"
+
+/// \brief fills uniformAltitudeDataM with altitude data which evenly distributed by
+/// |resultPointCount| points. |distanceDataM| and |altitudeDataM| form a curve of route altitude.
+/// This method is used to generalize and evenly distribute points of the chart.
+void NormalizeChartData(deque<double> const & distanceDataM,
```

Comment:
Thank you for detailed answer. In this case I don't want throw away the current content. I suggest to have a look at the code I'll push in a few minutes. Is it ok with you?